CaloOnlineTools/HcalOnlineDb/src/HcalLutAnalyzer.cc

```diff
@@ -0,0 +1,541 @@
+
+	    hlut[hbhe][2]->Fill(i, llut1[i]==0?0:(double)llut2[i]/llut1[i]);
+
+	    if(i==size-1 || (llut1[i]==maxvalue || llut2[i]==maxvalue)){ //Fill with only the last one before the maximum 
+		if(llut1[i-1]==0 || llut2[i-1]==0) { 
+		    break; 		
+		}
+		double condratio=lutgain[1][iraw]/lutgain[0][iraw] * lutresp[1][iraw]/lutresp[0][iraw]; 
+		xfill= (double)llut2[i-1]/llut1[i-1];
+		hslope[hbhe]->Fill(condratio, xfill);
+
+		h[4][0][hbhe][idep]->Fill(ieta, iphi, (double)llut1[i-1]/(i-1));
+		h[4][1][hbhe][idep]->Fill(ieta, iphi, (double)llut2[i-1]/(i-1));
+		h[4][2][hbhe][idep]->Fill(ieta, iphi, xfill);
+		r[4][hbhe]->Fill(ieta, xfill);
+		p[4][hbhe]->Fill(ieta, xfill);
+
+		break;
+	    }
+	}
+    }
+
+    gROOT->SetStyle("Plain");
+    gStyle->SetPalette(1);
+    gStyle->SetStatW(0.2);
+    gStyle->SetStatH(0.1);
+    gStyle->SetStatY(1.0);
+    gStyle->SetStatX(0.9);
+    gStyle->SetOptStat(110010);
+    gStyle->SetOptFit(111111);
+
+    //Draw and Print
+    TCanvas *cc = new TCanvas("cc", "cc", 0, 0, 1600, 1200);
+    cc->SetGridy();
+    for(int j=0; j<NVAR; ++j){
+	gSystem->mkdir(TString(plotsDir)+"/_"+titleVar[j]);
+	for(int d=0; d<NDET; ++d){
+	    cc->Clear(); 
+	    r[j][d]->Draw("colz"); 
+	    cc->Print(TString(plotsDir)+"/_"+titleVar[j]+"/"+TString(r[j][d]->GetName())+".pdf");
+
+	    cc->Clear(); 
+	    p[j][d]->Draw(); 
+	    if(doRatio[j]){
+		p[j][d]->SetMinimum(Pmin);
+		p[j][d]->SetMaximum(Pmax);
+	    }
+	    cc->Print(TString(plotsDir)+"/_"+titleVar[j]+"/"+TString(p[j][d]->GetName())+".pdf");
+
+	    for(int i=0; i<NLEV; ++i){
+		for(int p=0; p<DEP[d]; ++p){
+		    cc->Clear(); 
+		    h[j][i][d][p]->Draw("colz"); 
+
+		    if(i==NLEV-1){
+			if(doRatio[j]){
+			    h[j][2][d][p]->SetMinimum(Zmin);
+			    h[j][2][d][p]->SetMaximum(Zmax);
+			}
+			else{
+			    h[j][2][d][p]->SetMinimum(-3);
+			    h[j][2][d][p]->SetMaximum( 3);
+			}
+		    }
+		    cc->Print(TString(plotsDir)+"/_"+titleVar[j]+"/"+TString(h[j][i][d][p]->GetName())+".pdf");
+		}
+	    }
+	}
+    }
+
+    for(int i=0; i<NLEV; ++i){
+	cc->Clear(); 
+	hlut[0][i]->Draw(); 
+	hlut[1][i]->Draw("sames"); 
+	hlut[2][i]->Draw("sames"); 
+	hlut[3][i]->Draw("sames"); 
+	cc->Print(TString(plotsDir)+Form("LUT_%d.gif",i));
+    }
+    cc->Clear(); hslope[0]->Draw("colz"); cc->SetGridx(); cc->SetGridy(); cc->Print(TString(plotsDir)+"GainLutScatterHBHE.pdf");
+    cc->Clear(); hslope[1]->Draw("colz"); cc->SetGridx(); cc->SetGridy(); cc->Print(TString(plotsDir)+"GainLutScatterLutHF.pdf");
+
+    for(int i=0; i<2; ++i){
+	cc->Clear(); 
+	houtput[0][i]->Draw("box"); 
+	houtput[1][i]->Draw("samebox"); 
+	houtput[2][i]->Draw("samebox"); 
+	houtput[3][i]->Draw("samebox"); 
+	cc->Print(TString(plotsDir)+Form("OUT_%d.gif",i));
+    }
+}
+
+
+
+
+
+void
+HcalLutAnalyzer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+
+DEFINE_FWK_MODULE(HcalLutAnalyzer);
```

Comment:
Hi @akhukhun - CaloOnlineTools/HcalOnlineDb/src can either be a library or define plugins, but not both. You should probably move this class to the plugins directory.